### PR TITLE
Issue #9579: update example of AST for TokenTypes.SUPER_CTOR_CALL

### LINKED ISSUE #9579: update example of AST for TokenTypes.SUPER_CTOR_CALL
https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#SUPER_CTOR_CALL

needs to be extended to show example of AST.
Source to be updated: https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java

PLEASE READ for more details: https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-update-AST-in-TokenTypes

### PR DESCRIPTION
**Pull request by vivek286**
Closes #9579 

WhatsApp Image 2021-03-16 at 5 48 13 PM

Expected update example of AST for TokenTypes.SUPER_CTOR_CALL

WhatsApp Image 2021-03-16 at 5 47 16 PM

### CODE REVIEW

**Comment by vivek286** [comment]
@pbludov please review if you have time

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 1]
```diff
@@ -1067,20 +1067,17 @@
      *
      * <p>For example:</p>
      * <pre>
-     * this(1);
+     * |--this -&gt; 1
```
Comment (pbludov) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:1070]: Please revert this change. `this(1);` is the expression parsed for this example.

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 2]
```diff
@@ -1067,20 +1067,17 @@
...
-     *     |
-     *     +--LPAREN (()
-     *     +--ELIST
-     *         |
-     *         +--EXPR
-     *             |
-     *             +--NUM_INT (1)
-     *     +--RPAREN ())
-     *     +--SEMI (;)
+     * |--CTOR_CALL -&gt; CTOR_CALL
```
Comment (pbludov) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:1074]: Please remove all leftmost `|`.  The fragment should start from `CTOR_CALL`.
Since the `SEMI` token is a child of `CTOR_CALL`, no siblings of `CTOR_CALL` are expected here.

For cases like [this](https://github.com/checkstyle/checkstyle/pull/9431/files#diff-ba93830ba2cf23d542b30560fb43bdc8d8d1be05025e84ee6a96cca6ef8e3342) where `SEMI` is a sibling, the leftmost column with `|` is needed.

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 3]
```diff
@@ -1067,20 +1067,17 @@
...
+     * |    |--LPAREN -&gt; (
+     * |    |--ELIST -&gt; ELIST
+     * |        |--EXPR -&gt; EXPR
+     * |            |`--NUM_INT -&gt; 1
```
Comment (pbludov) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:1078]: Should be **`--NUM_INT**, this is the last child.

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 4]
```diff
@@ -1067,20 +1067,17 @@
...
+     * |    |--LPAREN -&gt; )
+     * |    |--SEMI -&gt; ;
```
Comment (pbludov) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:1080]: Should be **`--SEMI**, this is the last child.

**Review by pbludov — CHANGES_REQUESTED** [review]
Please do not craft this doc manually. You should create a test input and run
```bash
java -jar checkstyle-8.41-all.jar -J TestClass.java
````
then strip the ` [line:column]` from right and all irrelevant lines from left.

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java [hunk 5]
```diff
@@ -1067,17 +1067,17 @@
...
      * <p>parses as:</p>
      * <pre>
-     * |--CTOR_CALL -&gt; CTOR_CALL
-     * |    |--LPAREN -&gt; (
-     * |    |--ELIST -&gt; ELIST
-     * |        |--EXPR -&gt; EXPR
-     * |            |`--NUM_INT -&gt; 1
-     * |    |--LPAREN -&gt; )
-     * |    |--SEMI -&gt; ;
+     * --CTOR_CALL -&gt; CTOR_CALL
```
Comment (pbludov) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:1074]: One `|` is still needed:
```
CTOR_CALL -> this
 |--LPAREN -> (
 |--ELIST -> ELIST
 |   `--EXPR -> EXPR
```
...

**Review by pbludov — CHANGES_REQUESTED** [review]

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 5 above)
Comment (vivek286) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:1074]: Please guide me through the development IDE setup, the process given in documentation is a bit confusing.

**Comment by AkMo3** [comment]
@vivek286 you need to squash your commits into one. Please read [Beginning Development](https://checkstyle.org/beginning_development.html) , if still things remain unclear, you can refer from [here](https://stackoverflow.com/questions/5189560/squash-my-last-x-commits-together-using-git) .

**Comment by romani** [comment]
@vivek286 , there should be 1 commit in this PR, please squash commits.

File: src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java (hunk 5 above)
Comment (romani) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/TokenTypes.java:1074]: https://checkstyle.org/beginning_development.html

**Review by pbludov — APPROVED** [review]
I'm ok to merge. Please make CI green.

**Review by romani — CHANGES_REQUESTED** [review]
CI should be green
